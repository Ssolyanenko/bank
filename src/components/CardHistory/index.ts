export * from './History';
export * from './Pagination';
export * from './CardHistory';
