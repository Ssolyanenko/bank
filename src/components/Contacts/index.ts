export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
