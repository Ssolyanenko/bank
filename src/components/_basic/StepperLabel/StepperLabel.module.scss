@import 'src/styles/variables.module.scss';

.step {
  display: flex;
  width: 180px;
  align-items: center;
  justify-content: center;

  & .circle {
    width: 16px;
    height: 16px;
    border-radius: 50%;
    background-color: $grayLight_5;
  }

  & .connector {
    width: 100%;
    height: 4px;
    border-radius: 2px 0 0 2px;
    background-color: $grayLight_5;
  }

  .active {
    background-color: $orange;
  }
}
