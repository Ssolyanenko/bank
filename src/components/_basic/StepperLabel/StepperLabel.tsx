import { FC, ReactElement } from 'react';

import classes from './StepperLabel.module.scss';

interface Props {
  isCompleted: boolean;
}

export const StepperLabel: FC<Props> = ({ isCompleted }): ReactElement => {
  const active: string = isCompleted ? classes.active : '';

  return (
    <div className={classes.step}>
      <div className={`${classes.circle} ${active}`} />
      <div className={`${classes.connector} ${active}`} />
    </div>
  );
};
