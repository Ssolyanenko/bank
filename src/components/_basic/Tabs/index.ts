export * from './Tabs';
export * from './TabTitle';
