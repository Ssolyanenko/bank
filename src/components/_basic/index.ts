export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
