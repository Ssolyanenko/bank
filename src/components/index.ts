export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
