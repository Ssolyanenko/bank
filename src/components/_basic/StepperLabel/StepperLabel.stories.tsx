import { ComponentMeta, ComponentStory } from '@storybook/react';

import { StepperLabel } from './StepperLabel';

export default {
  title: 'Basic/Stepper/StepperLabel',
  component: StepperLabel,
  argTypes: {
    onClose: {
      open: false,
    },
  },
  parameters: {
    docs: {
      description: {
        component: 'Label of step to change password',
      },
    },
  },
} as ComponentMeta<typeof StepperLabel>;

const Template: ComponentStory<typeof StepperLabel> = (props) => <StepperLabel {...props} />;

export const Default = Template.bind({});

Default.args = {
  isCompleted: true,
};
