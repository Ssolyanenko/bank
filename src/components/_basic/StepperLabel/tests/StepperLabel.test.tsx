import React from 'react';
import { shallow } from 'enzyme';

import { StepperLabel } from 'components/_basic';
import { Default } from 'components/_basic/StepperLabel/StepperLabel.stories';

describe('<StepperLabel />', (): void => {
  test('should make a snapshot of component', (): void => {
    const props = {
      isCompleted: false,
    };
    const wrapper = shallow(<StepperLabel {...props} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('renders StepperLabel without crashing', (): void => {
    const props = {
      isCompleted: false,
    };
    const wrapper = shallow(<StepperLabel {...props} />);

    expect(wrapper.exists()).toBe(true);
  });

  test('should render StepperLabel from storybook', (): void => {
    const props = {
      isCompleted: false,
    };
    const wrapper = shallow(<Default {...props} />);

    expect(wrapper.find(StepperLabel).length).toBe(1);
  });
});
