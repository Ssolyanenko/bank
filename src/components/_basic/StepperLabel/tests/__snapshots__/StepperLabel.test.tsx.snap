// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepperLabel /> should make a snapshot of component 1`] = `
<div
  className="step"
>
  <div
    className="circle "
  />
  <div
    className="connector "
  />
</div>
`;
